### ISSUE TITLE
hack/download.sh using latest release page

### ISSUE THREAD

**Issue by StevenLi-phoenix** [fatedier/frp#4201]
### Describe the feature request

The code for [download.sh(https://github.com/fatedier/frp/blob/e81b36c5ba3ee0b4b88ed8832f4a302ed3289bda/hack/download.sh#L15) used the release page and do a search on that page.
Is it possible to use the latest release page to identify the version?

### Describe alternatives you've considered

Curl from url https://github.com/fatedier/frp/releases/latest instead of doing whole release page search.

### Affected area

- [ ] Docs
- [X] Installation
- [ ] Performance and Scalability
- [ ] Security
- [ ] User Experience
- [ ] Test and Release
- [ ] Developer Infrastructure
- [ ] Client Plugin
- [ ] Server Plugin
- [ ] Extensions
- [ ] Others

**Comment by fatedier** [comment]
No. Although it is correct most of the time, here only the official version number will be downloaded, excluding rc, beta, etc.

In addition, this script is only used for testing scenarios and is not available to the public. Please do not use it in a production environment as it may be modified or removed in the future.
